### PR TITLE
[Code Health] Replace deprecated SingleLiveEvent with Event<> wrapper

### PR DESCRIPTION
**Pull request by gino-m**

### CODE REVIEW

File: gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationViewModel.java [hunk 1]
```diff
@@ -241,14 +241,14 @@ private static boolean isAddObservationRequest(EditObservationFragmentArgs args)
...
+      return Single.just(Event.create(SaveResult.NO_CHANGES_TO_SAVE));
     }
     refreshValidationErrors();
     if (hasValidationErrors()) {
-      return Single.just(Event.of(SaveResult.HAS_VALIDATION_ERRORS));
+      return Single.just(Event.create(SaveResult.HAS_VALIDATION_ERRORS));
     }
     if (!hasUnsavedChanges()) {
-      return Single.just(Event.of(SaveResult.NO_CHANGES_TO_SAVE));
+      return Single.just(Event.create(SaveResult.NO_CHANGES_TO_SAVE));
```
[the remark this answers is not in the document]
Comment (gino-m) [reply gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationViewModel.java]: The issue with this approach is that the `Runner` would be associated with the subscriber, not the observer. Assuming it's the UI that's subscribing, if it get gets destroyed and recreated, when it resubscribes a new Runner will be created, and the event will trigger.

**Comment by gino-m** [comment]
> Looks great! I do wonder if we need both classes, or if we can just offload most of the complexity to one class that wraps Consumers and not have to worry about wrapping data in a corresponding special type.

This is what I had initially, but we ended up with the ugly `Event.create(Nil.NIL)`,  `Event<Nil>`, and an ignored arg in `ifUnhandled(__ -> ...)`. Stronger typing seems sightly less bad.

If we keep these, anything clearer than `Action` and `Event`? `HandleableAction` and `HandleableData` maybe? 🤷‍♂

**Comment by gino-m** [comment]
@scolsen  Could I get a final stamp if it LG?

**Comment by scolsen** [comment]
> > Looks great! I do wonder if we need both classes, or if we can just offload most of the complexity to one class that wraps Consumers and not have to worry about wrapping data in a corresponding special type.
> 
> This is what I had initially, but we ended up with the ugly `Event.create(Nil.NIL)`, `Event<Nil>`, and an ignored arg in `ifUnhandled(__ -> ...)`. Stronger typing seems sightly less bad.
> 
> If we keep these, anything clearer than `Action` and `Event`? `HandleableAction` and `HandleableData` maybe? 🤷‍♂

Ahh, gotcha. I think action and event are fine for now!

File: gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationViewModel.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (scolsen) [reply gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationViewModel.java]: Ah, thanks for the explanation!

**Pull request merged**